pynest/examples/Potjans_2014/network.py

```diff
@@ -0,0 +1,477 @@
+        self.K_scaling = self.net_dict['K_scaling']
+        self.synapses = get_total_number_of_synapses(self.net_dict)
+        self.synapses_scaled = self.synapses * self.K_scaling
+        self.nr_neurons = self.N_full * self.N_scaling
+        self.K_ext = self.net_dict['K_ext'] * self.K_scaling
+        self.w_from_PSP = get_weight(self.net_dict['PSP_e'], self.net_dict)
+        self.weight_mat = get_weight(
+            self.net_dict['PSP_mean_matrix'], self.net_dict
+            )
+        self.weight_mat_std = self.net_dict['PSP_std_matrix']
+        self.w_ext = self.w_from_PSP
+        if self.net_dict['poisson_input']:
+            self.DC_amp_e = np.zeros(len(self.net_dict['populations']))
+        else:
+            print(
+                '''
+                no poisson input provided
+                calculating dc input to compensate
+                '''
+                )
+            self.DC_amp_e = compute_DC(self.net_dict, self.w_ext)
+
+        print(
+            'The Number of neurons is scaled by a factor of: %.2f'
+            % self.N_scaling
+            )
+        print(
+            'The number of synapses is scaled by a factor of: %.2f'
+            % self.K_scaling
+            )
+
+        # Scaling of the synapses.
+        if self.K_scaling != 1:
+            synapses_indegree = self.synapses / (
+                self.N_full.reshape(len(self.N_full), 1) * self.N_scaling)
+            self.weight_mat, self.w_ext, self.DC_amp_e = adj_w_ext_to_K(
+                synapses_indegree, self.K_scaling, self.weight_mat,
+                self.w_from_PSP, self.DC_amp_e, self.net_dict, self.stim_dict
+                )
+
+        # Create cortical populations.
+        self.pops = []
+        pop_file = open(
+            os.path.join(self.data_path, 'population_GIDs.dat'), 'w+'
+            )
+        for i, pop in enumerate(self.net_dict['populations']):
+            population = nest.Create(
+                self.net_dict['neuron_model'], int(self.nr_neurons[i])
+                )
+            nest.SetStatus(
+                population, {
+                    'tau_syn_ex': self.net_dict['neuron_params']['tau_syn_ex'],
+                    'tau_syn_in': self.net_dict['neuron_params']['tau_syn_in'],
+                    'E_L': self.net_dict['neuron_params']['E_L'],
+                    'V_th': self.net_dict['neuron_params']['V_th'],
+                    'V_reset':  self.net_dict['neuron_params']['V_reset'],
+                    't_ref': self.net_dict['neuron_params']['t_ref'],
+                    'I_e': self.DC_amp_e[i]
+                    }
+                )
+            self.pops.append(population)
+            pop_file.write('%d  %d \n' % (population[0], population[-1]))
+        pop_file.close()
+        nest.sli_run(
+            '0 << /model /%s >> GetLocalNodes' % self.net_dict['neuron_model']
+            )
+        local_nodes = nest.sli_pop()
+        vp = nest.GetStatus(local_nodes)[0]['vp']
```

Comment:
This might be a possible solution to the problem:

```python
for thread in np.arange(nest.GetKernelStatus('local_num_threads')):
    local_nodes = nest.GetNodes([0], {'model': self.net_dict['neuron_model'], 'thread': thread}, local_only=True)[0]
    vp = nest.GetStatus(local_nodes)[0]['vp'] # same thread on MPI proc -> same vp
    nest.SetStatus(
        local_nodes, 'V_m', self.pyrngs[vp].normal(
        self.net_dict['neuron_params']['V0_mean'],
        self.net_dict['neuron_params']['V0_sd'],
            len(local_nodes)))
```